lambda/request-queue/src/main.ts

```diff
@@ -72,7 +95,15 @@ export const handler = async () => {
                 },
               ),
               createEvent({ eventCode: EVENTS.REQUEST_APPLY_FAILURE, requestId: request.id }),
-            ]);
+            ];
+            if (queuedRequest.attempts === MAX_ATTEMPTS - 1) {
```

Comment:
is there any way for async effects to cause this logic to be skipped? eg the queue jumps from max attemp-2 to maxattempt?

Reply:
I don't think so, but also no harm in it being >= just in case, since the function returns above at max. I will update it